src/backends/gr.jl

```diff
@@ -1103,7 +1103,16 @@ function gr_legend_pos(sp::Subplot, leg, vp)
         return gr_legend_pos(lp, vp)
     end
     if (leg_str = string(lp)) == "best"
-        leg_str = "topright"  # NOTE: can we do better auto-positioning, as `PyPlot` does ?
+        lp = _guess_best_legend_position(:best,sp) 
+        if lp == :topleft
+            leg_str = "topleft"
+        elseif lp == :bottomleft
+            leg_str = "bottomleft"
+        elseif lp == :bottomright
+            leg_str = "bottomright"
+        else lp == :topright
+            leg_str = "topright"
+        end
```

Comment:
What I meant was:

```suggestion
    leg_str = string(_guess_best_legend_position(lp, sp))
```

and remove `    if (leg_str = string(lp)) == "best"`.